### PR TITLE
Add support for Async operations on Worker

### PR DESCRIPTION
**Pull request by ocalvo**

### CODE REVIEW

**Comment by ocalvo** [comment]
> > Can you please add some tests for this code?
> 
> Certainly. Thanks for the quick reply.

How do you run tests locally?

**Comment by ocalvo** [comment]
> > > Can you please add some tests for this code?
> > 
> > 
> > Certainly. Thanks for the quick reply.
> 
> How do you run tests locally?

I am sorry, but looks like I wont be able to add tests, the test suite is running with python 2 and the async keyword is not supported:
```
Traceback (most recent call last):
  File "setup.py", line 240, in <module>
    ext_modules=[get_module()]
  File "C:\Python27\lib\site-packages\setuptools\__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "C:\Python27\lib\distutils\core.py", line 151, in setup
    dist.run_commands()
  File "C:\Python27\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "C:\Python27\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\Python27\lib\site-packages\setuptools\command\test.py", line 229, in run
    self.run_tests()
  File "C:\Python27\lib\site-packages\setuptools\command\test.py", line 251, in run_tests
    exit=False,
  File "C:\Python27\lib\unittest\main.py", line 94, in __init__
    self.parseArgs(argv)
  File "C:\Python27\lib\unittest\main.py", line 149, in parseArgs
    self.createTests()
  File "C:\Python27\lib\unittest\main.py", line 158, in createTests
    self.module)
  File "C:\Python27\lib\unittest\loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "C:\Python27\lib\unittest\loader.py", line 103, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "C:\Python27\lib\site-packages\setuptools\command\test.py", line 55, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "C:\Python27\lib\unittest\loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "c:\projects\python-gammu\test\test_asyncworker.py", line 34
    async def test_worker_async(self):
            ^
SyntaxError: invalid syntax
Command exited with code 1
```

**Comment by nijel** [comment]
It's probably time to drop Python 2 support, feel free to make tests Python 3 only.

**Comment by ocalvo** [comment]
> It's probably time to drop Python 2 support, feel free to make tests Python 3 only.

Sounds good to me.

**Comment by nijel** [comment]
I've just dropped Python 2 support in d181cb46c6f00f2bd2c621a16b7d409f7bbb8a8a

**Comment by ocalvo** [comment]
> I've just dropped Python 2 support in [d181cb4](https://github.com/gammu/python-gammu/commit/d181cb46c6f00f2bd2c621a16b7d409f7bbb8a8a)

Do I need another release for 3.1 or we can ship async with 3.0?

**Comment by nijel** [comment]
It will be with 3.0, there is no plan for it yet.

File: examples/async.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+'''
+python-gammu - Phone communication libary
+Gammu asynchronous wrapper example with asyncio. This allows your application to care
+only about handling received data and not about phone communication
+details.
+'''
+
+import sys
+import pprint
+pp = pprint.PrettyPrinter(indent=4)
```
Comment (nijel) [inline examples/async.py:31]: Seems like forgotten debugging code?

**Comment by ocalvo** [comment]
Scrutinizer failed with `The code coverage data was not received within the specified time. Please make sure your third-party service is configured to send code coverage data.`
Am I missing something?

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by ocalvo** [comment]
Thanks, when do you think you can release 3.0?
We need this to implement incoming SMS notifications in [Home Assistant](https://github.com/home-assistant/core/pull/34406)

**Comment by nijel** [comment]
I think I can do it now :-)

**Comment by ocalvo** [comment]
> I think I can do it now :-)

Awesome

**Pull request merged**